### PR TITLE
new artifact format supported by images API

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Changes:
- artifact library updated
- support for new artifact format

TODO:
- enhance swagger spec

Solves:
- MEN-734

### CODE REVIEW

File: docs/integrations_api.yml [hunk 1]
```diff
@@ -598,73 +598,109 @@ definitions:
...
+          type: array
+          items:
+            $ref: "#/definitions/UpdateFile"
+        meta_data:
+          type: object
+          description: |
+              meta_data is an object of unknown structure as this is dependent of update type (also custom defined by user)
+  ArtifactInfo:
+      description: |
+          Informations about artifact format and version.
```
Comment (bboozzoo) [inline docs/integrations_api.yml]: Information

File: resources/images/controller/images.go [hunk 2]
```diff
@@ -338,59 +363,56 @@ func (s *SoftwareImagesController) getMetaFromArchive(
 	rp := &parser.RootfsParser{}
 	aReader.Register(rp)
 
-	_, err := aReader.ReadInfo()
+	info, err := aReader.ReadInfo()
 	if err != nil {
 		return nil, errors.Wrap(err, "info error")
 	}
+	metaYocto.Info = getArtefactInfo(info)
```
Comment (bboozzoo) [inline resources/images/controller/images.go]: Maybe now is time to rename `*Yocto*` to `*Artifact*`. Same for `SoftwareImageMetaYoctoConstructor`.

File: resources/images/image.go [hunk 3]
```diff
@@ -41,22 +41,62 @@ func (s *SoftwareImageMetaConstructor) Validate() error {
...
+
+// Update structure
+type Update struct {
+	TypeInfo ArtifactUpdateTypeInfo `json:"type_info" valid:"required"`
+	Files    UpdateFiles            `json:"files"`
+	MetaData interface{}            `json:"meta_data" valid:"optional"` //TODO check this
+}
+
+// Information provided with YOCTO image
+type SoftwareImageMetaYoctoConstructor struct {
```
Comment (bboozzoo) [inline resources/images/image.go]: Rename to `SoftwareImageMetaArtifactConstructor`.

File: resources/images/image.go [hunk 4]
```diff
@@ -41,22 +41,62 @@ func (s *SoftwareImageMetaConstructor) Validate() error {
...
+	// artifact_name from artifact file
+	ArtifactName string `json:"artifact_name" valid:"length(1|4096),required"`
+
+	// Compatible device model for the application
+	DeviceTypesCompatible []string `json:"device_types_compatible" bson:"device_types_compatible" valid:"required"`
+
+	// Artifat version info
```
Comment (bboozzoo) [inline resources/images/image.go]: Artifact

File: resources/images/controller/images.go [hunk 5]
```diff
@@ -330,6 +331,30 @@ func (s *SoftwareImagesController) handleImage(
 	return tmpfile, meta, http.StatusOK, nil
 }
 
+func getArtefactInfo(info *metadata.Info) *images.ArtifactInfo {
```
Comment (bboozzoo) [inline resources/images/controller/images.go]: tomato, tomahto

You use artifact in other places, why make this func a special snowflake?

File: resources/images/image.go [hunk 6]
```diff
@@ -41,22 +41,62 @@ func (s *SoftwareImageMetaConstructor) Validate() error {
...
-	DateBuild *time.Time `json:"date_build" valid:"optional"`
+	Date *time.Time `json:"date" valid:"optional"`
+}
+
+type UpdateFiles []UpdateFile
+
+// Update structure
+type Update struct {
+	TypeInfo ArtifactUpdateTypeInfo `json:"type_info" valid:"required"`
+	Files    UpdateFiles            `json:"files"`
```
Comment (bboozzoo) [inline resources/images/image.go]: `[]UpdateFile` seems fine to me, at least it's clearly visible this field is a list of `UpdateFile`.

File: resources/images/controller/images.go (hunk 5 above)
Comment (kjaskiewiczz) [inline resources/images/controller/images.go]: done

File: resources/images/image.go (hunk 6 above)
Comment (kjaskiewiczz) [inline resources/images/image.go]: done

File: resources/images/image.go (hunk 4 above)
Comment (kjaskiewiczz) [inline resources/images/image.go]: done

File: resources/images/controller/images.go (hunk 2 above)
Comment (kjaskiewiczz) [inline resources/images/controller/images.go]: done

File: resources/images/image.go (hunk 3 above)
Comment (kjaskiewiczz) [inline resources/images/image.go]: done

**Comment by bboozzoo** [comment]
@mchalski you want to take a look?

**Pull request merged**